### PR TITLE
Add information icon to Proximal Table

### PR DESCRIPTION
**Pull request by UAMS16**
This hover feature should help clear up the meaning of " Proximal Regulatory Effect Observations". For example, icon hover displays "Nearby regulatory effect observations that have not been experimentally linked to LMO2"

Targeting Regulatory Effect Observations-For example, icon hover displays "Regulatory effect observations targeting LMO2"

Clarifying Dna features with help icon-Text reads "DNA features include regulatory elements, genes, exons, and transcripts."

image
image
image
image
image

### CODE REVIEW

File: cegs_portal/search/views/v1/reg_effects.py [hunk 1]
```diff
@@ -231,8 +239,9 @@ class TargetEffectsView(FeatureEffectsView):
     table_partial = "search/v1/partials/_target_reg_effect.html"
     tsv_renderer = target_data
 
-    def get_data(self, options, feature_id) -> tuple[QuerySet[RegulatoryEffectObservation], DNAFeature]:
+    def get_data(self, options, feature_id) -> tuple[QuerySet[RegulatoryEffectObservation], DNAFeature, DNAFeature]:
         feature = RegEffectSearch.id_feature_search(feature_id)
+        dna_feature = DNAFeature.objects.get(accession_id=feature_id)
```
Comment (tncowart) [inline cegs_portal/search/views/v1/reg_effects.py:244]: feature and dna_feature are the same thing.

```
    @classmethod
    def id_feature_search(cls, feature_accession_id: str):
        return DNAFeature.objects.get(accession_id=feature_accession_id)
```

You don't need to do a separate lookup.

File: cegs_portal/search/views/v1/reg_effects.py [hunk 2]
```diff
@@ -242,13 +251,13 @@ def get_data(self, options, feature_id) -> tuple[QuerySet[RegulatoryEffectObserv
                 feature_id, options.get("sig_only"), self.request.user.all_experiments()
             )
 
-        return reg_effects, feature
+        return reg_effects, feature, dna_feature
 
     def get_tsv(self, request, options, data, feature_id):
-        feature = DNAFeature.objects.get(accession_id=feature_id)
+        dna_feature = DNAFeature.objects.get(accession_id=feature_id)
```
Comment (tncowart) [inline cegs_portal/search/views/v1/reg_effects.py:257]: Huh. I'm curious why this was even here. Isn't this already in the "data" argument? I think you can take it out.

File: cegs_portal/search/views/v1/reg_effects.py [hunk 3]
```diff
@@ -178,24 +178,31 @@ def request_options(self, request):
...
     def get_json(self, request, options, data, *args, **kwargs):
-        regeffects, _ = data
+        regeffects, _, _ = data
         reg_effect_paginator = Paginator(regeffects, options["per_page"])
         reg_effect_page = reg_effect_paginator.get_page(options["page"])
 
         return super().get_json(request, options, reg_effect_page, *args, **kwargs)
 
-    def get_data(self, options, feature_id) -> tuple[QuerySet[RegulatoryEffectObservation], DNAFeature]:
+    def get_data(self, options, feature_id) -> tuple[QuerySet[RegulatoryEffectObservation], DNAFeature, DNAFeature]:
```
Comment (tncowart) [inline cegs_portal/search/views/v1/reg_effects.py:205]: You don't need the new DNAFeature return value.

**Review by tncowart — CHANGES_REQUESTED** [review]

File: cegs_portal/search/templates/search/v1/dna_feature.html [hunk 4]
```diff
@@ -286,6 +286,9 @@
                             d="M416 0c17.7 0 32 14.3 32 32c0 59.8-30.3 107.5-69.4 146.6c-28 28-62.5 53.5-97.3 77.4l-2.5 1.7c-11.9 8.1-23.8 16.1-35.5 23.9l0 0 0 0 0 0-1.6 1c-6 4-11.9 7.9-17.8 11.9c-20.9 14-40.8 27.7-59.3 41.5H283.3c-9.8-7.4-20.1-14.7-30.7-22.1l7-4.7 3-2c15.1-10.1 30.9-20.6 46.7-31.6c25 18.1 48.9 37.3 69.4 57.7C417.7 372.5 448 420.2 448 480c0 17.7-14.3 32-32 32s-32-14.3-32-32H64c0 17.7-14.3 32-32 32s-32-14.3-32-32c0-59.8 30.3-107.5 69.4-146.6c28-28 62.5-53.5 97.3-77.4c-34.8-23.9-69.3-49.3-97.3-77.4C30.3 139.5 0 91.8 0 32C0 14.3 14.3 0 32 0S64 14.3 64 32H384c0-17.7 14.3-32 32-32zM338.6 384H109.4c-10.1 10.6-18.6 21.3-25.5 32H364.1c-6.8-10.7-15.3-21.4-25.5-32zM109.4 128H338.6c10.1-10.7 18.6-21.3 25.5-32H83.9c6.8 10.7 15.3 21.3 25.5 32zm55.4 48c18.4 13.8 38.4 27.5 59.3 41.5c20.9-14 40.8-27.7 59.3-41.5H164.7z"/>
                     </svg>
                     Closest DNA Features
+                    <div class="ml-2">
+                        {% include 'search/v1/partials/_help_tooltip.html' with help_text="Closest DNA features to transcription start site(TSS). DNA features include regulatory elements, genes, exons, and transcripts." %}
```
[the remark this answers is not in the document]
Comment (UAMS16) [reply cegs_portal/search/templates/search/v1/dna_feature.html:290]: You mean in other tables?
image

[the remark this answers is not in the document]
Comment (tncowart) [reply cegs_portal/search/templates/search/v1/dna_feature.html:290]: Ah, okay. That's fine then. Just put a space before it.

**Review by tncowart — APPROVED** [review]

**Pull request merged**